classes/userPrivateNote/DAO.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * @file classes/userPrivateNote/DAO.php
```

Comment:
Rather than a bespoke table for this feature (with all the attendant management that'll require), another option would've been to use `NoteDAO` with an `assoc_type` of  `ASSOC_TYPE_CONTEXT`. Notes are currently used for a few things, and the assoc_type/assoc_id pairing is similar to a Laravel "morphs" relationship.

Reply:
Yes, I think on balance it'll be worth reusing the notes feature. That would be my preference, above e.g. moving to Laravel or sticking with DAOs. We already have >100 tables and all the attendant classes etc. to go with them, and it's a lot of overhead to maintain!